### PR TITLE
Update change-log.md

### PR DESCRIPTION
**Pull request by MartaD**

### CODE REVIEW

File: docs/change-log.md [hunk 1]
```diff
@@ -2,6 +2,16 @@
 
 This document provides an overview of the changes we are making over time. Feel free to reach out to us in case of any specific questions.
 
+## Product Update October 20th, 2023
+
+- **New:** The Humanitec Terraform Driver is now in General Availability! This major release allows more seamless integrations and management of infrastructure as code (IaC) with Terraform. Check out the [feature announcement](https://humanitec.com/blog/the-humanitec-terraform-driver-enters-general-availability) for more details.
```
Comment (sujaya-sys) [inline docs/change-log.md:7]:
```suggestion
- **New:** The Humanitec Terraform driver is now at general availability (GA). Following a successful experimental beta phase, we have addressed implementation gaps, resulting in a more robust and reliable user experience. Check out the [feature announcement](https://humanitec.com/blog/the-humanitec-terraform-driver-enters-general-availability) for more details.
```

Comment (chrishumanitec) [reply docs/change-log.md:7]: Suggest: "is now generally available" instead of "is now at general availability"

File: docs/change-log.md [hunk 2]
```diff
@@ -2,6 +2,16 @@
...
+
+- **Improved:** Lowered the latency on all authorization-related API calls through a new authorization management system, ensuring faster and more responsive interactions across our platform.
```
Comment (chrishumanitec) [inline docs/change-log.md:9]: Wasn't this announced before?

**Review by chrishumanitec — COMMENTED** [review]
Wording changes

File: docs/change-log.md (hunk 2 above)
Comment (MartaD) [reply docs/change-log.md:9]: No, it was released right after the last product update on September 15th.

**Review by sujaya-sys — APPROVED** [review]

**Pull request merged**